### PR TITLE
APPINT-32776: [7.3.1] tDbInput not working with dynamic type in schem…

### PR DESCRIPTION
**Pull request by vdrokov**
…a in runtime

**What is the current behavior?** (You can also link to an open issue here)

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: core/components-common/src/main/java/org/talend/components/common/config/jdbc/MappingFileLoader.java [hunk 1]
```diff
@@ -68,9 +69,27 @@
         return null;
     }
 
+    public List<Dbms> load(URL file) {
+        DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+
+        try {
+            DocumentBuilder analyser = documentBuilderFactory.newDocumentBuilder();
+            Document document = analyser.parse(file.openConnection().getInputStream());
```
Comment (clesaec) [inline core/components-common/src/main/java/org/talend/components/common/config/jdbc/MappingFileLoader.java:77]: risk that inputstream never close; better with 
```java
try (final InputStream input = file.openStream())  {
      DocumentBuilder analyser = documentBuilderFactory.newDocumentBuilder();
      Document document = analyser.parse(input);
```

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/796/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/796/)

**Review by clesaec — APPROVED** [review]
:+1:

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/795/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/795/)

File: core/components-common/src/main/java/org/talend/components/common/config/jdbc/MappingFileLoader.java [hunk 2]
```diff
@@ -68,9 +69,27 @@
...
+            NodeList dbmsNodes = document.getElementsByTagName("dbms");
+            return constructAllDbms(dbmsNodes);
+        } catch (ParserConfigurationException e) {
+            e.printStackTrace();
+        } catch (SAXException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```
[the remark this answers is not in the document]
Comment (vdrokov) [reply core/components-common/src/main/java/org/talend/components/common/config/jdbc/MappingFileLoader.java:87]: @undx , I've copied these lines from the above method which was there before me.
image

**Pull request closed without merging**